src/mpl/include/mpl_atomic.h

```diff
@@ -107,6 +107,10 @@ static void MPL_atomic_compiler_barrier(void);
 #elif defined(MPL_HAVE_NT_INTRINSICS)
 #include "mpl_atomic_nt_intrinsics.h"
 #elif defined(MPL_USE_LOCK_BASED_PRIMITIVES)
+#include <pthread.h>
+typedef pthread_mutex_t MPL_emulation_ipl_t;
+int MPL_atomic_interprocess_lock_init(MPL_emulation_ipl_t * shm_lock, int isLeader);
```

Comment:
To use it, we just need allocate `MPL_emulation_ipl_t mpl_atomic_lock` on shared memory, uninitialized, and have each process call `MPL_atomic_interprocess_lock_init(&mpl_atomic_lock, isLeader)`, right?

Reply:
Yes, you're right.